{{cookiecutter.project_dir}}/charts/{{cookiecutter.project_dir}}-api/templates/deployment.yaml

```diff
@@ -49,6 +49,10 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: status.podIP
+        - name: HOST_NAME
+          value: "{{ .Values.service.name }}{{ .Values.jxRequirements.ingress.namespaceSubDomain }}{{ .Values.jxRequirements.ingress.domain }}"
+        - name: CORS_ALLOWED_ORIGINS
+          value: "{{ .Values.service.baseName }}-ui{{ .Values.jxRequirements.ingress.namespaceSubDomain }}{{ .Values.jxRequirements.ingress.domain }}"
```

Comment:
It might be better to define a function in the helpers file. Also, you can leverage the built-in `quote` method.
I also think that this should go in the `values.yaml` since its specific to the use-case where UI and API app both exist and integrate.